aclk/mqtt.c

```diff
@@ -244,12 +249,41 @@ void _link_shutdown()
             break;
     };
 
+    if (netdata_exit)
+        return;
+
+    _link_event_loop();
+
     mosquitto_destroy(mosq);
     mosq = NULL;
 
     aclk_lws_wss_client_destroy();
 }
 
+
+int _link_set_lwt(char *sub_topic, int qos)
+{
+    int rc;
+    char topic[ACLK_MAX_TOPIC + 1];
+    char payload[sizeof(ACLK_LWT_MSG)+128+1];
+    char *final_topic;
+
+    final_topic = get_topic(sub_topic, topic, ACLK_MAX_TOPIC);
+    if (unlikely(!final_topic)) {
+        errno = 0;
+        error("Unable to build outgoing topic; truncated?");
+        return 1;
+    }
+
+    time_t time_created = now_realtime_sec();
+    char *msg_id = create_uuid();
+    snprintfz(payload, sizeof(ACLK_LWT_MSG)+128, ACLK_LWT_MSG, msg_id, time_created, "unexpected");
```

Comment:
This should be `sizeof(payload)` to guard against changes at either site in the future.

Reply:
Removed the define